### ISSUE TITLE
RFE: port libyang to pcre2

### ISSUE THREAD

**Issue by kloczek** [CESNET/libyang#1383]
On Linux most of teh core system libraries like SELinux libraries are usisng pcre2.
It would be good to reuse the same ABI as that one which definitelly is always used on each Linux.

**Comment by kloczek** [comment]
Just found sssd sample patch which could be usefull on porting to pcre2 https://patch-diff.githubusercontent.com/raw/SSSD/sssd/pull/677.patch

**Comment by kloczek** [comment]
And another one for php https://patch-diff.githubusercontent.com/raw/php/php-src/pull/2857.patch

**Comment by michalvasko** [comment]
*libyang* version 2, which we are still finishing up, uses *pcre2*. The current *libyang* will keep using *pcre*, its development is finished, we are only providing bugfixes.

**Comment by kloczek** [comment]
Thx for explanation :)
Is that new version will be API compliant with last tagged version?

**Comment by rkrejci** [comment]
No, principles are very close, but API changed. There is a [transition manual](https://netopeer.liberouter.org/doc/libyang/libyang2/html/transition.html) describing the differences and we appreciate any feedback for the new API.
